### PR TITLE
perf: Addressing timeouts

### PR DESCRIPTION
**Pull request by HarisSQA**
Speed up wallet-selector. Delete unnecessary timeouts and remove waitFor function where it don't needed.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change.

- [ ] FIX - a PR of this type patches a bug.
- [ ] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [x] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

File: packages/nearfi/src/lib/nearfi.ts [hunk 1]
```diff
@@ -240,9 +240,7 @@ export function setupNearFi({
 
     // Add extra wait to ensure NearFi's sign in status is read from the
     // browser extension background env.
-    await waitFor(() => !!window.nearFiWallet?.isSignedIn(), {
-      timeout: 300,
-    }).catch(() => false);
+    await waitFor(() => !!window.nearFiWallet?.isSignedIn()).catch(() => false);
```
[the remark this answers is not in the document]
Comment (HarisSQA) [reply packages/nearfi/src/lib/nearfi.ts:243]: Agree, I will remove it.

File: packages/nearfi/src/lib/nearfi.ts [hunk 2]
```diff
@@ -240,9 +239,6 @@ export function setupNearFi({
 
     // Add extra wait to ensure NearFi's sign in status is read from the
     // browser extension background env.
```
Comment (kujtimprenkuSQA) [inline packages/nearfi/src/lib/nearfi.ts:241]: Can you please remove this comment too?

**Review by kujtimprenkuSQA — COMMENTED** [review]
What was the outcome of the research for `detectEthereumProvider` was it reasonable to add that `timeout` option to reduce the waiting time (for NETH)?

File: packages/nearfi/src/lib/nearfi.ts (hunk 2 above)
Comment (HarisSQA) [reply packages/nearfi/src/lib/nearfi.ts]: Removed.

**Review by DamirSQA — APPROVED** [review]

**Pull request merged**